Administrator Guide/Performance Tuning.md

```diff
@@ -0,0 +1,70 @@
+# Performance tuning
+
+## Enable Metadata cache
+Metadata caching improves performance in almost all the workloads, except for use cases
+with most of the workload accessing a file sumultaneously from multiple clients.
+ 1. Execute the following command to enable metadata caching and cache invalidation:
+    ```
+    # gluster volume set <volname> group metadata-cache
+    ```
+    This group command enables caching of stat and xattr information of a file or directory.
+    The caching is refreshed every 10 min, and cache-invalidation is enabled to ensure cache
+    consistency.
+
+ 2. To increase the number of files that can be cached, execute the following command:
+    ```
+    # gluster volume set <volname> network.inode-lru-limit <n>
+    ```
+    n, is set to 50000. It can be increased if the number of active files in the volume
+    is very high. Increasing this number increases the memory footprint of the brick processes.
+
+ 3. Execute the following command to enable samba specific metadata caching:
+    ```
+    # gluster volume set <volname> cache-samba-metadata on
+    ```
+
+ 4. By default, some xattrs are cached by gluster like: camability xattrs, ima xattrs
+    ACLs, etc. If there are any other xattrs that are used by the application using
+    the Gluster storage, execute the following command to add these xattrs to the metadata
+    cache list:
+    ```
+    # gluster volume set <volname> xattr-cache-list "comma separated xattr list"
+    ```
+    Eg:
+    ```
+    # gluster volume set <volname> xattr-cache-list "user.org.netatalk.*,user.swift.metadata"
+    ```
+
+## Directory operations
+Along with enabling the metadata caching, the following options can be set to increase directory operations:
+   ### Directory listing Performance:
+   1. Enable Parallel-readdir
```

Comment:
This isn't rendered correctly. You can lose the bullet numbering to fix it.
Same comment below for nl-cache commands.